### PR TITLE
Dtos and handlers fixes, create order mechanism updated, formatting

### PR DESCRIPTION
**Pull request by BobAesthetic**

### CODE REVIEW

File: backend/src/Core/FurnitureShop.Core.Contracts/Mobile/ShoppingCart/ShoppingCartDTOs.cs [hunk 1]
```diff
@@ -8,11 +7,9 @@ public class ShoppingCartDTO
     {
         public double Price { get; set; }
         public IEnumerable<ShoppingCartProductDTO> ShoppingCartProducts { get; set; }
-
     }
-    public class ShoppingCartProductDTO
+    public class ShoppingCartProductDTO : ProductDTO
     {
         public int Amount { get; set; }
-        public ProductDTO Product { get; set; }
```
Comment (aliaksei-liavonik) [inline backend/src/Core/FurnitureShop.Core.Contracts/Mobile/ShoppingCart/ShoppingCartDTOs.cs:16]: keep it here to not have in the same level amout of products and product

File: backend/src/Core/FurnitureShop.Core.Contracts/Web/Products/ProductDTOs.cs [hunk 2]
```diff
@@ -1,35 +1,23 @@
...
+    public class ProductDetailsDTO : ProductDTOBase
     {
-        public string Name { get; set; }
-        public double Price { get; set; }
-        public double? AverageRating { get; set; }
-        public string? PreviewPhotoId { get; set; }
-        public Guid? CategoryId { get; set; }
+        public Guid Id { get; set; }
+        public string? ModelId { get; set; }
+        public List<Guid> PhotosIds { get; set; }
```
Comment (aliaksei-liavonik) [inline backend/src/Core/FurnitureShop.Core.Contracts/Web/Products/ProductDTOs.cs:11]: PhotoIds

File: backend/src/Core/FurnitureShop.Core.Services/CQRS/Mobile/Favourites/MyFavouritesQH.cs [hunk 3]
```diff
@@ -66,19 +65,18 @@ private async Task<List<Guid>> GetProductsInShoppingCart(CoreContext context)
...
+                .Where(shp => shp.ShoppingCartId == shoppingCartId.Value)
                 .Select(shp => shp.ProductId!.Value).ToListAsync();
         }
 
         private async Task<List<Guid>> GetProductsInFavourites(CoreContext context)
         {
             return await dbContext.Favourites
-                .Where(f => f.UserId == context.UserId.Value && f.ProductId != null)
+                .Where(f => f.UserId == context.UserId.Value)
                 .Select(f => f.ProductId!.Value).ToListAsync();
```
Comment (aliaksei-liavonik) [inline backend/src/Core/FurnitureShop.Core.Services/CQRS/Mobile/Favourites/MyFavouritesQH.cs:76]: Why it is nullable

File: backend/src/Core/FurnitureShop.Core.Services/CQRS/Mobile/Orders/CreateOrderCH.cs [hunk 4]
```diff
@@ -54,29 +72,30 @@ public CreateOrderQH(CoreDbContext dbContext)
...
-                    OrderedDate = DateTime.Now,
-                    OrderState = OrderState.Pending,
+                    Amount = prod.Amount,
+                    ProductId = prod.ProductId,
+                    OrderId = newOrder.Id
                 });
-            if (result.Entity != null)
+            }
+            var result = await dbContext.Orders.AddAsync(newOrder);
+            if(result != null)
```
Comment (aliaksei-liavonik) [inline backend/src/Core/FurnitureShop.Core.Services/CQRS/Mobile/Orders/CreateOrderCH.cs:94]: I already created a layout for taking a specific amount of specific products in the shopping cart, maybe revert this somehow?

Comment (BobAesthetic) [reply backend/src/Core/FurnitureShop.Core.Services/CQRS/Mobile/Orders/CreateOrderCH.cs:94]: ok zrewertuje

File: backend/src/Core/FurnitureShop.Core.Services/CQRS/Mobile/Favourites/MyFavouritesQH.cs (hunk 3 above)
Comment (BobAesthetic) [reply backend/src/Core/FurnitureShop.Core.Services/CQRS/Mobile/Favourites/MyFavouritesQH.cs:76]: po zmianach w enitity framwork już nie jest nullable

File: backend/src/Core/FurnitureShop.Core.Contracts/Mobile/Orders/OrderDTOs.cs [hunk 5]
```diff
@@ -15,14 +17,9 @@ public class OrderDTOBase
...
-    public class OrderDTO : OrderDTOBase
-    {
-        public Guid Id { get; set; }
-        public List<ProductInOrderDTO> Products { get; set; }
-    }
-
-    public class CreateOrderDTO : OrderDTOBase
+    public class CreateOrderDTO
     {
+        public string Address { get; set; }
```
Comment (aliaksei-liavonik) [inline backend/src/Core/FurnitureShop.Core.Contracts/Mobile/Orders/OrderDTOs.cs:22]: string? Address

**Review by aliaksei-liavonik — APPROVED** [review]

**Pull request merged**